crates/backend/src/encode.rs

```diff
@@ -83,6 +85,7 @@ impl Interner {
             let msg = "relative module paths aren't supported yet";
             return Err(Diagnostic::span_error(span, msg))
         } else {
+            self.check_for_package_json();
```

Comment:
I think we want to unconditionally check for `package.json` all the time. The npm package might put stuff on the global scope (generally not recommended, but might be required for polyfills or something). This would mean that the use of the dependency is effectively invisible to us, and we can't assume that if there is no module import, then we can avoid the declared npm dependency.

Reply:
Sounds reasonable to me!